### DISCUSSION TITLE
More context for disconnects from multiplayer matches

### DISCUSSION THREAD

**Discussion by hburn7** [ppy/osu#18486] [Ideas] [answered]
For competitive matches, it can be helpful to know whether a player left the lobby deliberately or if they disconnected due to a network issue. I'm not sure if this is feasible but there are tournaments right now that have rules against intentionally disconnecting due to poor gameplay or otherwise, in the hopes of getting another chance to play that map. Right now, referees only know that a player has left - they don't have any other context from the client or chat as to why they left.

The main caveat to this idea is if someone's game crashed, that is a hard failure and is usually not intentional, but to the game it's seen as the player leaving normally. So, I think knowing whether a player is experiencing a network issue of some sort would be sufficient.

**Comment by peppy** [comment, marked as answer]
poor connection should not disconnect. this is covered by other issues and something we are working to fix.
